eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -196,10 +196,11 @@ extends:
               ArtifactPath: "$(System.DefaultWorkingDirectory)/$(BreakingChangeLabelArtifactPath)"
               CustomCondition: and(succeeded(), ne(variables['BreakingChangeLabelArtifactName'], ''))
 
-          - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
-            parameters:
-              ArtifactName: $(PipelineArtifactsName)
-              ArtifactPath: $(GeneratedSDK.StagedArtifactsFolder)
+          - ${{ if variables['GeneratedSDK.StagedArtifactsFolder'] }}:
```

Comment:
This will not work. At the time of evaluation it's just the literal string `variables['GeneratedSDK.StagedArtifactsFolder']` which is always truthy.

Unfortunately you can't use most variables at template parameter compile time. You'll have to pass this down via `CustomCondition` instead (https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/steps/publish-1es-artifact.yml#L13).